core/src/main/scala-2.12/cats/data/ChainCompanionCompat.scala

```diff
@@ -0,0 +1,42 @@
+package cats.data
+
+import cats.data.Chain.{nil, one, Wrap}
+import cats.kernel.compat.scalaVersionSpecific.IterableOnce
+
+import scala.collection.immutable.{Seq => ImSeq}
+import scala.collection.mutable.{Seq => MutSeq}
+
+private[data] trait ChainCompanionCompat {
+
+  /**
+   * Creates a Chain from the specified sequence.
+   */
+  def fromSeq[A](s: Seq[A]): Chain[A] =
+    s match {
+      case mut: MutSeq[A] => fromMutableSeq(mut)
+      case imm: ImSeq[A]  => fromImmutableSeq(imm)
```

Comment:
actually, `Seq` isn't sealed, so in principle there could be other subclasses.

Can we instead do:
```scala
s match {
  case imm: ImSeq[A] => fromImmutableSeq(imm)
  case _ =>
    if (s.isEmpty) Empty
    else if (s.lengthCompare(1) == 0) Singleton(s.head)
    else fromImmutableSeq(s.toVector)
}
```